src/main/java/org/sciborgs1155/lib/NoteVisualizer.java

```diff
@@ -0,0 +1,72 @@
+package org.sciborgs1155.lib;
+
+import static org.sciborgs1155.robot.pivot.PivotConstants.*;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.math.geometry.Translation3d;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.ScheduleCommand;
+import java.util.Set;
+import java.util.function.DoubleSupplier;
+import java.util.function.Supplier;
+import monologue.Annotations.Log;
+import org.sciborgs1155.robot.Constants.Field;
+import org.sciborgs1155.robot.pivot.PivotConstants;
+
+public class NoteVisualizer {
+  @Log.NT static Pose3d[] notes = new Pose3d[] {};
+  static Supplier<Pose2d> pose = () -> new Pose2d();
+  static Supplier<Rotation2d> angle = () -> PivotConstants.STARTING_ANGLE;
+  static DoubleSupplier velocity = () -> 1;
+
+  public static void setSuppliers(
+      Supplier<Pose2d> robotPose, Supplier<Rotation2d> pivotAngle, DoubleSupplier shotVelocity) {
+    pose = robotPose;
+    angle = pivotAngle;
+    velocity = shotVelocity;
+  }
+
+  public static Command shoot() {
```

Comment:
for this command, you don't want to use the example method of interpolating between poses.
instead you should actually simulate the note as a projectile
this means storing a current velocity for the note, and updating it with acceleration * dt every tick
also, you only want to delete the current note when it hits the floor, not every note